perun-core/src/main/java/cz/metacentrum/perun/core/entry/UsersManagerEntry.java

```diff
@@ -362,9 +362,12 @@ public UserExtSource updateUserExtSource(PerunSession sess, UserExtSource userEx
 			throw new PrivilegeException(sess, "updateUserExtSource");
 		}
 
-		getUsersManagerBl().checkUserExtSourceExists(sess, userExtSource);
+		if (getUsersManagerBl().checkUserExtSourceExistsById(sess, userExtSource.getId())) {
```

Comment:
Check methods should throw exception instead of returning value.

Reply:
Reworked to throw exception instead of returning boolean. Rest of comments is not relevant then.